pre-cache/parse_index.py

```diff
@@ -27,54 +24,46 @@ def parse_args():
     return args
 
 
-def extract_bundle_names(args):
+def extract_images(args, objects):
     bundles = []
     with open(args.operators_spec_file.name, 'r') as p:
         records = [i.split(":") for i in p.read().splitlines() if len(i) > 0]
+    packages = []
+    channels = {}
     for item in records:
-        pkg_name = item[0].strip()
-        path = os.path.join(args.index_download_path, pkg_name, 'package.yaml')
-        with open(path, 'r') as f:
-            package = yaml.safe_load(f)
-            csv_name = [p.get("currentCSV") for p in
-                        package.get("channels")
-                        if p.get("name") ==
-                        item[1].strip()][0].strip(f"{pkg_name}.")
-        with os.scandir(os.path.join(args.index_download_path, pkg_name)) as it:
-            bundle_dirs = [entry.name for entry in it if entry.is_dir()]
-        bundle_dir = sorted(
-            bundle_dirs, key=lambda dir_name: SequenceMatcher(
-                None, csv_name, dir_name).ratio())[-1]
-        bundles.append(os.path.join(
-            args.index_download_path, pkg_name, bundle_dir))
-    return bundles
-
-
-def extract_images(bundles):
+        channels[item[0].strip()] = item[1].strip()
+        packages.append(item[0].strip())
```

Comment:
Maybe save item[0].strip() in a variable, since you are referencing them twice consecutively.